libsql.go

```diff
@@ -772,9 +786,37 @@ Outerloop:
 }
 
 func (c *conn) QueryContext(ctx context.Context, query string, args []sqldriver.NamedValue) (sqldriver.Rows, error) {
-	rowsNativePtr, err := c.execute(query, args, false)
+	rowsNativePtr, err := c.executeContext(ctx, query, args, false)
 	if err != nil {
 		return nil, err
 	}
 	return newRows(rowsNativePtr)
 }
+
+// doContext runs a blocking function f in a goroutine, and exits early if the given context is
+// canceled or if its deadline is exceeded.
+func doContext[U any](ctx context.Context, f func() (U, error)) (U, error) {
+	var res U
+	resCh := make(chan U)
+	errCh := make(chan error)
+
+	go func() {
+		defer close(resCh)
```

Comment:
This goroutine will be leaked forever if the doContext returns due to the context timout.
After doContext returns no one is receiving anything on `resCh` and `errCh` channels so when this go routine finishes executing `f()` it will try to send something either to `errCh` or `resCh` and will block because those are unbuffered channels. It will block forever.

Reply:
We probably need to make `resCh` and `errCh` buffered like `make(chan U, 1)`

Reply:
@haaawk you're right, good catch. I updated the commit.